inv/inventory.go

```diff
@@ -161,7 +179,14 @@ func (i *inventory) ReplaceAttributes(ctx context.Context, id model.DeviceID, up
 			}
 		}
 	}
-	if _, err := i.db.UpsertRemoveDeviceAttributes(ctx, id, upsertAttrs, removeAttrs); err != nil {
+
+	_, err = i.db.UpsertRemoveDeviceAttributes(ctx, id, upsertAttrs, removeAttrs, scope, etag)
+	if err != nil {
+		if scope == model.AttrScopeTags {
+			if strings.Contains(err.Error(), "E11000 duplicate key error collection") {
```

Comment:
@tranchitella @kjaskiewiczz I'm unsure about good enough productivity in this approach. Any ideas to make it more efficient?